### ISSUE TITLE
[Phase II] <Sinica CITI Boy> — <TASK 2> Technical Report

### ISSUE THREAD

**Issue by Allenchou0708** [EBiM-Benchmark/submissions#31]
### Team name

Sinica CITI Boy

### Point of Contact email

30033allen@gmail.com

### Task

Task 2 — Deformable Material Handling (Thermal Pad Placement)

### Your assigned Phase II pathway for this task

Munich / Augsburg

### Public GitHub repository URL

https://github.com/Allenchou0708/EBIM-Official-Repo.git

### Pinned commit SHA

405df4151663fe8ea95706a98b16ec4789f09876 in task2_phase2_real_pi05_hybrid_20260830 branch

### Build and run commands

Clone and check out the exact submitted revision:

git clone https://github.com/Allenchou0708/EBIM-Official-Repo.git
cd EBIM-Official-Repo
git checkout --detach <FINAL_PHASE2_COMMIT_SHA>

Build the policy image from the repository root:

docker build --pull \
  --file task2_real/Dockerfile \
  --tag academia-sinica-task2-phase2:<FINAL_PHASE2_COMMIT_SHA> \
  .

Optional container health check:

docker run --rm \
  --gpus all \
  academia-sinica-task2-phase2:<FINAL_PHASE2_COMMIT_SHA> \
  health

Launch the policy:

docker run --rm \
  --gpus all \
  --network host \
  --ipc host \
  --env ROS_DOMAIN_ID=<ORGANIZER_ASSIGNED_DOMAIN_ID> \
  --env RMW_IMPLEMENTATION=rmw_fastrtps_cpp \
  academia-sinica-task2-phase2:<FINAL_PHASE2_COMMIT_SHA> \
  run

The Docker image entrypoint is:

ENTRYPOINT ["/submission-entrypoint.sh"]

The `run` entrypoint command starts the complete policy in this order:

base navigation -> spine staging -> left-arm safe hold ->
right-arm observation pose -> right-arm/right-gripper PI0.5 inference.

The checkpoint, tokenizer, calibration schema, Python dependencies, and ROS 2
runtime are included in the image. No source-directory, dataset, checkpoint,
or cache bind mount is required. No Internet access is required at run time.
The only required external interface is the organizer's local ROS 2 network.

### Environment and dependencies

Linux x86_64 environment. The policy container is based on
nvidia/cuda:12.8.1-cudnn-runtime-ubuntu24.04 and uses Python 3.12,
PyTorch 2.11.0+cu128, cuDNN, a pinned LeRobot/PI0.5 implementation, and
ROS 2 Jazzy with rmw_fastrtps_cpp.

The host must provide Docker Engine, NVIDIA Container Toolkit, and an NVIDIA
driver compatible with CUDA 12.8. NVIDIA driver 580.126.18 has been validated;
we recommend driver 570 or newer. The container uses host networking and the
ROS_DOMAIN_ID assigned to the robot. The host does not otherwise require a
separate Python or machine-learning environment. Sufficient shared memory must
be available; our runtime uses --ipc=host.

The trained checkpoint, tokenizer, ROS interface configuration, and site
calibration profile must be included in the image or mounted read-only before
startup. The training dataset is not required at run time.

No Internet or external cloud access is required at run time. The only network
connectivity required is the local ROS 2 DDS network connecting the container
to the robot, cameras, lidar, and controllers.

### Hardware assumptions

The policy requires an NVIDIA CUDA GPU with at least 16 GB of VRAM; 24 GB or
more is preferred. It has been validated on an NVIDIA GeForce RTX 5090 with
32 GB VRAM. A pre-run inference and memory check is required if a different GPU
is used.

The real robot control stream is expected to run at 20 Hz, matching the released
demonstration data. In simulator validation, commands were executed at 30 Hz.
On the RTX 5090, one PI0.5 50-action chunk takes approximately 0.44–0.58 seconds
after warm-up. We use receding-horizon execution and commit at most 15 actions
per inference call while the independent safety supervisor continues monitoring
the robot. Three warm-up inferences should be completed before actuation.

At the beginning of each round, the robot must be in the organizer’s
collision-free reset pose. The base route and final table-relative pose must
be calibrated for each evaluation site; simulator world coordinates are not
used on the real robot. The deterministic controller then settles the base,
raises the spine to the site-calibrated height, places the left arm in a
collision-checked safe hold with its gripper open, and moves the right arm to
the calibrated observation/pre-grasp pose before handing control to PI0.5.

The thermal pad, source fixture, and target RAM/PCB must be restored to the
official Task 2 starting configuration, within the reachable workspace and
visible from the head and right-wrist cameras. The policy assumes normal,
approximately stable indoor illumination comparable to the released training
data, without severe glare, flicker, saturation, or camera occlusion.

Between rounds, staff must restore the pad and target fixture, return the base,
spine, both arms, and both grippers to the official initial state, clear any
dropped or deformed pad, reset controller faults, and confirm fresh camera,
joint, gripper, TF, odometry, lidar, torque, and wrench data. Exactly one command
publisher must own each actuator topic before the policy is armed.

### Does your policy rely on externally provided object poses?

No

### Did you train on the organizer-released trajectory dataset?

Yes — organizer-released trajectories only

### What changed since your Phase I submission? (optional)

Since the Phase I submission, we have explored several approaches to improve the policy, including PPO-based reinforcement learning, ACT-based imitation learning, human-skill prompt learning, and self-distillation.

### Submission requirements

- [x] The repository is public and the pinned commit is reachable in it.
- [x] It contains a Dockerfile that builds the image named in the run commands above.
- [x] It contains a README explaining how to run the policy on the real testbed.
- [x] We have run the commands above from a clean checkout of the pinned commit, and they work.

### Optional supplementary links

https://huggingface.co/Allenchou/EBIM_task2_model_academia_sinica_boys_v1_30000/tree/main

### Notes (optional)

We submit the Technical Report for Phase 2 competition, the report (Technical_Report_Sinica_CITI_Boy_Task2.md) is in the root repository in the github (https://github.com/Allenchou0708/EBIM-Official-Repo.git).  

### Acknowledgement

- [x] We understand organizers evaluate the pinned commit exactly as submitted, and will not modify our code to make it build or run.
- [x] We understand the operator has full authority to stop a run at any time, for safety or to protect the hardware.
- [x] We understand that if organizers cannot build and launch our submission from the fields above, it cannot be scored.

**Comment by yitongz328-hash** [comment]
After following the Phase II build instructions, I found that both main (28eada9) and the Phase II real-robot branch (1ad8966) are missing task2_real/Dockerfile. The existing task2_submission still uses the Phase I simulation submission entry point.
Please provide a FINAL_PHASE2_COMMIT_SHA that has been pushed to the remote repository, ensuring that it includes the Phase II Dockerfile, the real-robot policy entry point, and the model resources required for the build. Alternatively, please provide the published Phase II image address and digest.
The current instructions cannot be reproduced from the remote repository.

**Comment by yitongz328-hash** [comment]
we have found some same setup issues from the programms in Shanghai. For reference, we followed a similar process with EDL Team: after we reported practical deployment issues and requested adaptations, they revised their submission. The updated submission is available in https://github.com/EBiM-Benchmark/submissions/issues/41, and the detailed adaptation discussion can be found in the comments of https://github.com/EBiM-Benchmark/submissions/issues/39. EDL Team’s algorithm has now been successfully deployed on our testbed.

https://github.com/EBiM-Benchmark/submissions/issues/33 has also the similar issues. Hope you can change your programm
